spec/features/dashboard/collection_spec.rb

```diff
@@ -301,6 +301,51 @@
     end
   end
 
+  describe 'remove works from collection' do
+    context 'user that can edit' do
+      let!(:work2) { create(:work, title: ["King Louie"], member_of_collections: [collection1], user: user) }
+      let!(:work1) { create(:work, title: ["King Kong"], member_of_collections: [collection1], user: user) }
+
+      before do
+        sign_in admin_user
+      end
+      it 'shows remove action buttons' do
+        visit "/dashboard/collections/#{collection1.id}"
+        expect(page).to have_selector('input.collection-remove', count: 2)
+      end
+      it 'removes the first work from the list of items' do
+        visit "/dashboard/collections/#{collection1.id}"
+        expect(page).to have_selector('input.collection-remove', count: 2)
+        page.all('input.collection-remove')[0].click
+        expect(page).to have_selector('input.collection-remove', count: 1)
+        expect(page).not_to have_content(work1.title.first)
+        expect(page).to have_content(work2.title.first)
+      end
+      it 'removes the second work from the list of items' do
+        visit "/dashboard/collections/#{collection1.id}"
+        expect(page).to have_selector('input.collection-remove', count: 2)
+        page.all('input.collection-remove')[1].click
+        expect(page).to have_selector('input.collection-remove', count: 1)
+        expect(page).to have_content(work1.title.first)
+        expect(page).not_to have_content(work2.title.first)
+      end
+      xit 'removes a sub-collection from the list of items (dependency on collection nesting)' do
+      end
+    end
+    context 'user that cannot edit' do
```

Comment:
Can these be part of a view spec? This is much more of a unit-type test of a view. The feature tests are slow running compared to view specs.

Reply:
I first tried to write this as a view test, but the view relies on Ability and in that testing context it wasn't available and I couldn't figure out how to make it available. 
```Devise::MissingWarden:
       Devise could not find the `Warden::Proxy` instance on your request environment.
       Make sure that your application is loading Devise and Warden as expected and that the `Warden::Manager` middleware is present in your middleware stack.
       If you are seeing this on one of your tests, ensure that your tests are either executing the Rails middleware stack or that your tests are using the `Devise::Test::ControllerHelpers` module to inject the `request.env['warden']` object for you.```